#UDP sonnectionless error exception handling.
#except socket.timeout : = occures if the timeout is reached.
#except socket.herror : = address related error.
#except socket.gaierror : = catches  any exception that is raised due to getaddrinfo() and getnameinfo().
#except socket.error : = catches any socket related error.
import socket
s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
try:
	host = "127.0.0.1"
	port = 12345
	s.bind((host,port))
	s.settimeout(5)
	data, addr = s.recvfrom(1024)
	print "Connected to ", addr
	print "Obtained ", data
	s.close()
except socket.timeout :
	print "client not connected"
	s.close()

